core/block_render_svg_vertical.js

```diff
@@ -707,6 +729,29 @@ Blockly.BlockSvg.prototype.renderFields_ =
     root.setAttribute('transform',
       'translate(' + translateX + ', ' + translateY + ') ' + scale
     );
+
+    // If the first field is an image, as in extension blocks, and this field is the first field,
```

Comment:
Extract the code inside the `if` into a helper function.

Reply:
All set